aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -201,11 +353,31 @@ void max_pool2d_with_indices_out_cuda_template(
       scalar_t *input_data = input.data<scalar_t>();
       int64_t *indices_data = indices.data<int64_t>();
 
-      MaxPoolForward<scalar_t, scalar_t>
-        <<<cuda::ATenCeilDiv(count, num_threads), num_threads, 0, at::cuda::getCurrentCUDAStream()>>>(
-          count, input_data,
-          nbatch, nInputPlane, inputHeight, inputWidth, outputHeight, outputWidth,
-          kH, kW, dH, dW, padH, padW, dilationH, dilationW, output_data, indices_data); }
+      if (memory_format == MemoryFormat::ChannelsLast) {
+        const int count = safe_downcast<int, int64_t>(nbatch * outputHeight * outputWidth);
+        int* maxThreadsDim = at::cuda::getCurrentDeviceProperties()->maxThreadsDim;
+        int block_x = std::min<int>(maxThreadsDim[0], std::min<int>(lastPow2(count), at::cuda::warp_size()));
```

Comment:
We are doing a std::min here.
So likely you are only launching <= 32 threads per block. That will give you very low occupancy here.